### ISSUE TITLE
552 Mail with no Date header not accepted

### ISSUE THREAD

**Issue by reetp** [RocketChat/Rocket.Chat#3165]
Your Rocket.Chat version: 0.29.0
Linux Koozali SME Server v9 (CentOS 6)

Trying to simplify the install we have a few issues. The server is RFC compliant and quite strict about that.

Rocket Emails are sent without a date header. I have a small fix which seems to work - this adds the current time to the date header (excuse the code - I don't do JS !)

```
diff -ruN ./Rocket.Chat/programs/server/npm/email/node_modules/mailcomposer/lib/mailcomposer.js.old ./Rocket.Chat/programs/server/npm/email/node_modules/mailcomposer/lib/mailcomposer.js
--- ./Rocket.Chat/programs/server/npm/email/node_modules/mailcomposer/lib/mailcomposer.js.old   2016-05-17 08:03:02.000000000 +0200
+++ ./Rocket.Chat/programs/server/npm/email/node_modules/mailcomposer/lib/mailcomposer.js   2016-05-09 17:54:50.000000000 +0200
@@ -99,6 +99,15 @@
     this._gencounter = 0;

     this.addHeader("MIME-Version", "1.0");
+//  Set up the date
+    var dateObj = new Date();
+//  Convert to UTC
+    dateObj = dateObj.toUTCString();
+//  Now add the Date header element
+    this.addHeader("Date", dateObj);
+    
+//    var TestDate = require('child_process').execSync('date', {encoding: 'utf8'} )
+//    console.log(date);
 };

 /* PUBLIC API */

```

Two other issues ( I can open them as separate issues if required)

Subject line in default email has an unrequired `<br>`:

Subject: You have been added to Rocket.Chat`<br>`

Also noted that on initial install no 'From' header is added which causes the initial mail to fail to send via the local SMTP server.

I was also trying to find where the default email 'From' address is set in the DB and wondered if it was possible to set this prior to first running of Rocket ?

**Comment by reetp** [comment]
No fix for this ?

**Comment by reetp** [comment]
If you want to set Mongo DB settings and you can't wait for a fix then you can do this manually - if I knew more about JS I'd fix I'd propose a patch but.....:

Use mongo from the CLI:

 mongo
 use rocketchat

Look for some settings:

 db.rocketchat_settings.find({"_id" : "SMTP_Host"})
 db.rocketchat_settings.find({"_id" : "From_Email"})

Or more directly:

 db.rocketchat_settings.findOne({_id : "From_Email"}, {_id:0, value: 1})
 db.rocketchat_settings.findOne({_id : "SMTP_Host"}, {_id:0, value: 1})

Use this to update values:

 db.rocketchat_settings.update({"_id" : "From_Email"}, {$set: {"value":"admin@yourdomain.com"}})
 db.rocketchat_settings.update({"_id" : "SMTP_Host"}, {$set: {"value":"localhost"}})

Or from bash directly:

 mongo rocketchat --eval 'db.rocketchat_settings.update({"_id" : "SMTP_Host"}, {$set: {"value":"localhost"}});'
 mongo rocketchat --eval 'db.rocketchat_settings.update({"_id" : "From_Email"}, {$set: {"value":"admin@yourdomain.com"}});'

We can check the individual values set like this:

 mongo rocketchat --eval 'db.rocketchat_settings.find({"_id":"From_Email"}, {_id:0, value:1}).shellPrint();'
 mongo rocketchat --eval 'db.rocketchat_settings.find({"_id":"SMTP_Host"}, {_id:0, value: 1}).shellPrint();'

All values per _id:
 mongo rocketchat --eval 'db.rocketchat_settings.find({"_id" : "SMTP_Host"}).shellPrint();'
 mongo rocketchat --eval 'db.rocketchat_settings.find({"_id" : "From_Email"}).shellPrint();'

Only took me half a day to figure this lot out :-) The fix can't be that hard can it ?

**Comment by spinza** [comment]
Our mail is also blocking.  How can we add the date header to the emails.

**Comment by spinza** [comment]
I tried the fix above and it seems to have resolved it.  No more bad header errors.

Dug a little deeper and that code is actually out of mailcomposer, which is being used by Meteor Email (package?).  

mailcomposer is not setting this date automatically and it seems to be advised to set it manually by them.  A bit strange to not set a date as required by RFC but anyway:
https://github.com/nodemailer/mailcomposer/issues/16

So it needs to be set by Meteor Email as a header: https://docs.meteor.com/api/email.html#Email-send
The options sent to Email.send needs to include dictionary of "optional" headers. This should be a Date header.  

This is discussed in Meteor forums as well: https://forums.meteor.com/t/email-flagged-as-spam/5576/8

Places this code is affected are listed below.  It would seem far simpler to fix the Meteor or mailcomposer package upstream.

https://github.com/RocketChat/Rocket.Chat/blob/435fcef5dc68d156ac3e7d8c7587dbab6a1eeafe/packages/rocketchat-smarsh-connector/server/functions/sendEmail.js#L25

https://github.com/RocketChat/Rocket.Chat/blob/3425986902508acb6ca2a37be3c012a159e83231/packages/rocketchat-lib/server/methods/sendSMTPTestEmail.coffee#L15

https://github.com/RocketChat/Rocket.Chat/blob/3425986902508acb6ca2a37be3c012a159e83231/packages/rocketchat-mailer/server/functions/sendMail.coffee#L34

https://github.com/RocketChat/Rocket.Chat/blob/3425986902508acb6ca2a37be3c012a159e83231/packages/rocketchat-mailer/server/functions/sendMail.coffee#L34

https://github.com/RocketChat/Rocket.Chat/blob/056ca4908863779cdba69d7e63ffb06352ed741b/packages/rocketchat-lib/server/methods/sendInvitationEmail.coffee#L32

https://github.com/RocketChat/Rocket.Chat/blob/cd3778d0a645e1de1b735982d1368f40f047f132/packages/rocketchat-livechat/server/methods/sendOfflineMessage.js#L30

https://github.com/RocketChat/Rocket.Chat/blob/3425986902508acb6ca2a37be3c012a159e83231/packages/rocketchat-channel-settings-mail-messages/server/methods/mailMessages.coffee#L52

https://github.com/RocketChat/Rocket.Chat/blob/8839ac18689b1c285f615538457b2b8e689d3f95/packages/rocketchat-lib/server/lib/sendEmailOnMessage.js#L119

https://github.com/RocketChat/Rocket.Chat/blob/25ffa1919ddc26939cbd6be5f0985f133d0d0404/packages/rocketchat-lib/server/functions/saveUser.js#L113

**Comment by spinza** [comment]
Sorry for the long writing.  Now I see this commit:
https://github.com/RocketChat/Rocket.Chat/commit/b2afdb3af21ebecf9238896dd455fb7c37b947c6
In my noob Meteor experience this seems to include the fix that adds a Date header if none are present when Meteor sends an email. (This PR:https://github.com/meteor/meteor/pull/6916)

**Comment by reetp** [comment]
For a quick test I removed my original patch and installed 0.48 as unpatched. I sent a test email and did not see any adverse effects and the mail was received. I believe that the date header issue is fixed.

I am not sure if the default email 'From' address is set in the DB on install as per my comment above

https://github.com/RocketChat/Rocket.Chat/issues/3165#issuecomment-239965343

Needs a test with a fresh install to see.

**Comment by spinza** [comment]
Should be fixed since 0.48.1:
https://github.com/RocketChat/Rocket.Chat/commit/b2afdb3af21ebecf9238896dd455fb7c37b947c6#diff-b538af6434936bcdfd1833d3be563886
